src/chain/header.cpp

```diff
@@ -41,7 +41,7 @@ using wall_clock = std::chrono::system_clock;
 //-----------------------------------------------------------------------------
 
 header::header()
-  : header(0, null_hash, null_hash, 0, 0, 0)
+  : header(0, null_hash, null_hash, 0, 0, 0, settings{})
```

Comment:
`header::header()` is used by the factory functions below. I added this dummy value such that we can have the discussion here based on the code. I'd fix this by adding a `settings` parameter to this constructor and the factory functions. @evoskuil what do you think?

Reply:
Yes, require the parameter until all other repos have been updated, so that you do not end up missing the necessary settings passage. But at that point you can also add a defaulted construction.

Reply:
Alright, I have propagated the `settings` as a parameter wherever needed. This blew up the PR to some extend. Now, I know what you meant by this refactoring being "pervasive" :) Pushing new version now.